Tools/AP_Periph/can.cpp

```diff
@@ -1671,8 +1671,39 @@ void AP_Periph_FW::esc_telem_update()
     }
 }
 #endif // HAL_WITH_ESC_TELEM
-#endif // HAL_PERIPH_ENABLE_RC_OUT
 
+#ifdef HAL_PERIPH_ENABLE_ESC_APD
+void AP_Periph_FW::apd_esc_telem_update()
+{
+    for(uint8_t i = 0; i < ARRAY_SIZE(apd_esc_telem); i++) {
+        ESC_APD_Telem esc = apd_esc_telem[i];
```

Comment:
did you mean this to be a reference? It seems to be copying a complex class on the stack

Reply:
would be good to declare the class with CLASS_NO_COPY to prevent this

Reply:
Fixed.